pandas/tests/test_reshape.py

```diff
@@ -171,6 +172,26 @@ def test_basic(self):
         expected.index = list('ABC')
         assert_frame_equal(get_dummies(s_series_index, sparse=self.sparse), expected)
 
+    def test_basic_types(self):
+        # GH 10531
+        s_list = list('abc')
+        s_series = Series(s_list)
+        s_df = DataFrame({'a': [0, 1, 0, 1, 2],
+                          'b': ['A', 'A', 'B', 'C', 'C'],
+                          'c': [2, 3, 3, 3, 2]})
+
+        exp_type = DataFrame if not self.sparse else SparseDataFrame
+        self.assertEqual(type(get_dummies(s_list, sparse=self.sparse)), exp_type)
+        self.assertEqual(type(get_dummies(s_series, sparse=self.sparse)), exp_type)
+
+        r = get_dummies(s_df, sparse=self.sparse, columns=s_df.columns)
+        self.assertEqual(type(r), exp_type)
+
+        r = get_dummies(s_df, sparse=self.sparse, columns=['a'])
+        has_sparse_blocks = any(type(b) == pd.core.internals.SparseBlock
```

Comment:
ok, then test for the correct number / positioning of blocks. It is not obvious from just looking what you are testing.